File: YJFX/Text.py
# -*- code=utf-8 -*-
# Version 2.01  2019-04-23  Text クラス廃止
from typing import List, Tuple
import re

# 文字列のリスト
StrList = List[str]
StrTuple = Tuple[str]

# 文字列の長さ
def length(s:str) -> int:
  return len(s)

# 文字列の結合
def concat(str1:str, str2:str) -> str:
  return str1 + str2

# start から長さ length の部分文字列を返す。
def substring(text:str, start:int, length:int=-1) ->str:
  if length < 0 :
    return text[start:len(text)]
  else :
    return text[start:start+length]

# start から end までの部分文字列を返す。
def substr(text:str, start:int, end:int) ->str:
  return text[start:end+1]

# 先頭から長さ length の部分文字列を返す。
def left(text, length:str) -> str:
  return text[0 : length]

# 最後から長さ length の部分文字列を返す。
def right(text:str, length:int) -> str:
  n = len(text)
  begin = n - length
  return text[begin : n]

# 文字列 c を n 回繰り返した文字列をこのオブジェクトの値とする。
def times(c:str, n:int) -> str:
  return c * n

# 数字を判別
def isdigit(a: str) -> str:
  return a.isdigit()

# アルファベットを判別
def isalpha(a: str) -> str:
  return a.isalpha()

# 区切りを判別
def isdelim(a: str) -> str:
  return not a.isalnum()

# 表示可能かを判別
def isprint(a:str) ->bool:
  return a.isprintable()

# 小文字に変換
def tolower(s:str) -> bool:
  return s.lower()

# 大文字に変換
def toupper(s: str) -> bool:
  return s.upper()

# 前後の空白を取り除く
def trim(s: str) -> str:
  return s.strip()

# 後ろのCRやLFを取り除く
def chomp(s: str) -> str:
  if len(s) < 2 :
    return s
  last = len(s) - 1
  if s[last] == '\n' or s[last] == '\r' :
    s = s[0:last]
    last -= 1
    if s[last] == '\r' :
      s = s[0:last]
  return s

# 文字 c で文字列 s を分割してリストとして返す。
def split(c:bytes, s:str) -> StrList:
  return s.split(c)

# リスト array の要素を文字 c で連結する。
def join(c:str, array:list) -> str:
  strarray = []
  for a in array :
    if a == None :
      strarray.append("")
    else :
      strarray.append(str(a))
  return c.join(strarray)

# 文字列 s の中に部分文字列 p が含まれているか判別する。
def contain(p:str, s:str) -> bool:
  return (p in s)

# 文字列 s の中に部分文字列 p が含まれていればその位置を返す。
def indexOf(p:str, s:str) -> int:
  return s.find(p)

# 文字列 s に部分文字列 old が含まれていれば、それを new に置き換える。
def replace(old:str, new:str, s:str) -> str:
  return s.replace(old, new)

# 書式を含む文字列 form を *args (可変個数のパラメータ) で置き換える。
def format(form:str, *args: StrTuple) :
 return form.format(*args)

# 文字列 s を整数に変換する。
def parseInt(s:str) -> int:
 return int(s)

# 文字列 s を浮動小数点数に変換する。
def parseDouble(s:str) -> float:
 return float(s)

# 256未満の整数を文字に変換する。
def char(n:int) -> str:
 return chr(n)

# 文字を256未満の整数に変換する。
def asc(a:str) -> int:
  return ord(a)

#    正規表現を使う静的メソッド
# 正規表現 rstr が文字列 s に含まれるか判別する。
def re_contain(rstr:str, s:str) -> bool :
  ro = re.compile(rstr)
  m = ro.match(s)
  return m != None

# 正規表現 rstr が文字列 s に含まれていれば、その一致した部分文字列の集合を返す。
def re_search(rstr:str, s:str) -> re.match :
  ro = re.compile(rstr)
  m = ro.search(s)
  return m

# 正規表現 rstr が文字列 s に含まれていれば、その部分文字列で分割してリストとして返す。
def re_split(rstr:str, s:str) -> StrList :
  m = re.split(rstr, s)
  return m

# 正規表現 rstr が文字列 s に含まれていれば、その部分文字列を c で置き換える。
def re_replace(rstr:str, c:str, s:str) -> str :
  ro = re.compile(rstr)
  return ro.sub(c, s)

# 数(整数または浮動小数点数) d に3桁ごとにカンマを挿入した文字列を返す。
def money(d:float) -> str :
  m = '{0:,}'.format(d)
  return m

# bytes(バイト列)を文字列に変換する。
def b2s(b: bytes) -> str :
  return b.decode()

# 文字列(utf-8)をバイト列に変換する。
def s2b(s: str) -> bytes :
  return s.encode()

File: Filer2/Text.php
<?php
#   Text.php Version 1.00  2019-03-25
namespace Text;

# Text モジュール

# ブール値を文字列に変換する。
function bool2str(bool $b) {
  if ($b) {
    return "TRUE";
  }
  else {
    return "FALSE";
  }
}

# 文字 c が数字かどうか判別
function isdigit(string $c) : bool {
  $u = ord($c);
  if ($u >= 0x30 && $u <= 0x39) {
    return TRUE;
  }
  else {
    return FALSE;
  }
}

# 文字 c が英字かどうか判別
function isalpha(string $c) : bool {
  $u = ord($c);
  if (($u >= ord('A') && $u <= ord('Z')) ||($u >= ord('a') && $u <= ord('z'))) {
    return TRUE;
  }
  else {
    return FALSE;
  }
}

# 文字 c が区切りかどうか判別
function isdelim(string $c) : bool {
  if (isalpha($c)) {
    return FALSE;
  }
  else if (isdigit($c)) {
    return FALSE;
  }
  else if (ord($c) >= 0x20 && ord($c) <= 0x7f) {
    return TRUE;
  }
  else {
    return FALSE;
  }
}

# 文字 c が表示可能かどうか判別
function isprint(string $c) : bool {
  $u = ord($c);
  if ($u < 0x20) {
    return TRUE;
  }
  else {
    return FALSE;
  }
}

# 英小文字を大文字に変換する。
function tolower(string $text) : string {
  return strtolower($text);
}

# 英大文字を小文字に変換する。
function toupper(string $text) : string {
  return strtoupper($text);
}

# ASCII 文字列の長さを返す。
function len(string $text) : int {
  return strlen($text);
}

# 変数の型が文字列かどうか判別する。
function isstr($v) : bool {
  return is_string($v);
}

# 変数の型が整数かどうか判別する。
function isint($v) : bool {
  return is_int($v);
}

# 変数の型が数値かどうか判別する。
function isnum($v) : bool {
  return is_numeric($v);
}

# 文字列１の後に文字列２を追加する。
function append(string $text1, string $text2) : string {
  return $text1.$text2;
}

# ASCII 文字列 text の位置 start から長さ len の部分文字列を返す。
function substring(string $text, int $start, int $len) : string {
  return substr($text, $start, $len);
}

# ASCII 文字列 text の位置 start から位置 end までの部分文字列を返す。
function slice(string $text, int $start, int $end) : string {
  $len = $end - $start + 1;
  return substr($text, $start, $len);
}

# ASCII 文字列 text の長さ len の左側の部分文字列を返す。
function left(string $text, int $len) : string {
  return substr($text, 0, $len);
}

# ASCII 文字列 text の長さ len の右側の部分文字列を返す。
function right(string $text, int $len) : string {
  $start = strlen($text) - $len;
  return substr($text, $start, $len);
}

# 文字 c を n 回繰り返した文字列を返す。
function times(string $c, int $n) : string {
  $buff = '';
  for ($i = 0; $i < $n; $i++) {
    $buff .= $c;
  }
  return $buff;
}

# 引数を書式化した文字列を返す。
function format(string $fmt, ...$vars) : string {
  return sprintf($fmt, ...$vars);
}

# 引数を金額とし３桁おきにカンマを挿入した文字列を返す。
function money(float $val) : string {
  return number_format($val);
}

# 文字列 text に部分文字列 pat が含まれていれば TRUE, そうでなければ FALSE を返す。
function contains(string $text, string $pat) : bool {
  if (strpos($text, $pat) === FALSE) {
     return FALSE;
  }
  else {
     return TRUE;
  }
}

# 文字列 text に部分文字列 pat が含まれていればその位置を返す。含まれていない場合は負数を返す。
function indexOf(string $text, string $pat) : int {
   if (($p = strpos($text, $pat)) === FALSE) {
     return -1;
   }
   else {
      return $p;
   }
}

# 文字列 text の前後に空白があれば取り除く。
//function trim(string $text) : string {
//  return trim($text);
//}

# 文字列 text に部分文字列 pat が含まれていれば、すべて str に置き換える。
function replace(string $text, string $pat, string $str) : string {
  return str_replace($pat, $str, $text);
}

# 文字列 text に文字列 delim が含まれていれば、delim で区切った文字列に分割し、配列として返す。
function split(string $text, string $delim) : array {
  return explode($delim, $text);
}

# 文字列の配列 arr の要素をすべて文字列 delim で結合した文字列を返す。
//function join(array $arr, string $delim) : string {
//
//}

?>
